src/fileConverter.ts

```diff
@@ -219,67 +217,40 @@ export class FileConverter {
+     * @param upgradeOptions Controls how the user is prompted for SARIF log upgrades.
      */
-    public static async upgradeSarif(sarifFile: vscode.Uri, sarifVersion: SarifVersion | undefined, sarifSchemaVersion: SarifVersion | undefined, options: UpgradeSarifOptions): Promise<vscode.Uri | undefined> {
-
-        const saveTempChoice: vscode.MessageItem = {
-            title: localize('converterTool.Upgrade.SaveTemp', "Yes (Save Temp)")
-        };
+    public static async upgradeSarif(sarifFile: vscode.Uri, sarifVersion: SarifVersion | undefined, sarifSchemaVersion: SarifVersion | undefined, upgradeOptions?: UpgradeSarifOptions): Promise<vscode.Uri | undefined> {
+        const desiredUpgradeOption: UpgradeSarifOptions = upgradeOptions ?? vscode.workspace.getConfiguration(Utilities.configSection).get(FileConverter.UpgradePromptSettingName, DefaultUpgradeOption);
 
-        const saveAsChoice: vscode.MessageItem = {
-            title: localize('converterTool.Upgrade.SaveAs', "Yes (Save As)")
-        };
-
-        const noChoice: vscode.MessageItem = {
-            title: localize('converterTool.Upgrade.No', "No")
-        };
-
-        let choice: vscode.MessageItem | undefined;
+        if (desiredUpgradeOption === 'Prompt') {
+            const upgradeMessage: string = sarifSchemaVersion ?
+             localize('converterTool.Upgrade.AskWithSchema', "Sarif schema version {0} is not supported. Upgrade to the latest schema version {1}?", sarifSchemaVersion.original, FileConverter.MultiToolCurrentSchemaVersion.original) :
+             localize('converterTool.Upgrade.AskWithVersion', "Sarif version {0} is not supported. Upgrade to the latest version {1}?",
+                    sarifVersion && sarifVersion.original ? sarifVersion.original : localize('converterTool.Upgrade.UnknownVersion', "Unknown"), FileConverter.MultiToolCurrentSchemaVersion.original);
 
-        if (options.promptUserForUpgrade) {
-            let upgradeMessage: string;
+            const upgradeMessageItem: vscode.MessageItem = {
+                title: localize('converterTool.Upgrade.Upgrade', "Upgrade")
+            };
 
-            if (sarifSchemaVersion) {
-                upgradeMessage = localize('converterTool.Upgrade.AskWithSchema', "Sarif schema version {0} is not supported. Upgrade to the latest schema version {1}?", sarifSchemaVersion.original, FileConverter.MultiToolCurrentSchemaVersion.original);
-            } else {
-                upgradeMessage = localize('converterTool.Upgrade.AskWithVersion', "Sarif version {0} is not supported. Upgrade to the latest version {1}?",
-                    sarifVersion && sarifVersion.original ? sarifVersion.original : localize('converterTool.Upgrade.UnknownVersion', "Unknown"), FileConverter.MultiToolCurrentSchemaVersion.original);
-            }
+            const neverShowAgainMessageItem: vscode.MessageItem = {
+                title: localize('converterTool.Upgrade.NeverShow', "Don't Show Again")
+            };
```

Comment:
maybe the options can be "Yes" "Always" and "Never"?

Reply:
It's not always obvious that "Don't show again" means no... Although I suppose it is in this case, so you can take or leave that suggestion

---
In reply to: [416045107](https://github.com/microsoft/sarif-vscode-extension/pull/268#discussion_r416045107) [](ancestors = 416045107)

Reply:
I do think more clarity is needed. Just for context (and not justification), the we were trying to be more consistent with phrasing elsewhere in VS Code such as here:

![Dont Show Again](https://user-images.githubusercontent.com/3452831/80418334-1b1e7000-888c-11ea-9090-cb38b59740ab.png)

Reply:
So I also interpreted "Don't show again" meant as "Don't upgrade".

I need a moment to develop my opinion on this design.